hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneManagerLock.java

```diff
@@ -202,6 +196,26 @@ private boolean lock(Resource resource, String resourceName,
     }
   }
 
+  private void updateLockMetrics(Resource resource, String resourceName,
+                         Consumer<String> lockFn, String lockType) {
+    long startWaitingTimeNanos = Time.monotonicNowNanos();
+    lockFn.accept(resourceName);
+
+    /**
+     *  read/write lock hold count helps in metrics updation only once in case
+     *  of reentrant locks.
+     */
+    if (lockType.equals(READ_LOCK) &&
```

Comment:
Can we make this more readable, something like to use a switch case:

```
if (omLockMetricsCollectionEnabled) {
  switch (lockType) {
  case READ_LOCK:
    updateReadLockMetrics(resource, resourceName, startWaitingTimeNanos);
    break;
  case WRITE_LOCK:
    updateWriteLockMetrics(resource, resourceName, startWaitingTimeNanos);
    break;
  default:
    break;
  }
} else {
  lockFn.accept(resourceName);
}
```

Reply:
Thank you for the suggestion, I have made the required changes to incorporate the switch case.